patches/api/0418-More-MemoryKey-API.patch

```diff
@@ -0,0 +1,225 @@
++    public static final MemoryKey<Boolean> LONG_JUMP_MID_JUMP = create("long_jump_mid_jump", Boolean.class); // CraftBukkit forgot to add this one themselves
++    public static final MemoryKey<Vector> RAM_TARGET = create("ram_target", Vector.class);
++    public static final MemoryKey<Void> IS_IN_WATER = create("is_in_water", Void.class);
++    public static final MemoryKey<Void> IS_PREGNANT = create("is_pregnant", Void.class);
++    public static final MemoryKey<Boolean> IS_PANICKING = create("is_panicking", Boolean.class);
++    public static final MemoryKey<List<UUID>> UNREACHABLE_TONGUE_TARGETS = create("unreachable_tongue_targets", new TypeToken<List<UUID>>() {});
++    public static final MemoryKey<Integer> TIME_TRYING_TO_REACH_ADMIRE_ITEM = create("time_trying_to_reach_admire_item", Integer.class);
++    public static final MemoryKey<Boolean> DISABLE_WALK_TO_ADMIRE_ITEM = create("disable_walk_to_admire_item", Boolean.class);
++    public static final MemoryKey<Vector> CELEBRATE_LOCATION = create("celebrate_location", Vector.class);
++    public static final MemoryKey<Boolean> DANCING = create("dancing", Boolean.class);
++    public static final MemoryKey<Hoglin> NEAREST_VISIBLE_HUNTABLE_HOGLIN = create("nearest_visible_huntable_hoglin", Hoglin.class);
++    public static final MemoryKey<Hoglin> NEAREST_VISIBLE_BABY_HOGLIN = create("nearest_visible_baby_hoglin", Hoglin.class);
++    public static final MemoryKey<HumanEntity> NEAREST_TARGETABLE_PLAYER_NOT_WEARING_GOLD = create("nearest_targetable_player_not_wearing_gold", HumanEntity.class);
++    public static final MemoryKey<List<PiglinAbstract>> NEARBY_ADULT_PIGLINS = create("nearby_adult_piglins", new TypeToken<List<PiglinAbstract>>() {});
++    public static final MemoryKey<List<PiglinAbstract>> NEAREST_VISIBLE_ADULT_PIGLINS = create("nearest_visible_adult_piglins", new TypeToken<List<PiglinAbstract>>() {});
++    public static final MemoryKey<List<Hoglin>> NEAREST_VISIBLE_ADULT_HOGLINS = create("nearest_visible_adult_hoglins", new TypeToken<List<Hoglin>>() {});
++    public static final MemoryKey<PiglinAbstract> NEAREST_VISIBLE_ADULT_PIGLIN = create("nearest_visible_adult_piglin", new TypeToken<PiglinAbstract>() {});
++    public static final MemoryKey<LivingEntity> NEAREST_VISIBLE_ZOMBIFIED = create("nearest_visible_zombified", LivingEntity.class);
++    public static final MemoryKey<Integer> VISIBLE_ADULT_PIGLIN_COUNT = create("visible_adult_piglin_count", Integer.class);
++    public static final MemoryKey<Integer> VISIBLE_ADULT_HOGLIN_COUNT = create("visible_adult_hoglin_count", Integer.class);
++    public static final MemoryKey<HumanEntity> NEAREST_PLAYER_HOLDING_WANTED_ITEM = create("nearest_player_holding_wanted_item", HumanEntity.class);
++    public static final MemoryKey<Boolean> ATE_RECENTLY = create("ate_recently", Boolean.class);
++    public static final MemoryKey<Vector> NEAREST_REPELLENT = create("nearest_repellent", Vector.class); // TODO Position API
```

Comment:
This api now exists, so we can use it ?

Reply:
Oh you’re right. I totally forgot about transitioning that over to Position. Will do